### PR TITLE
Convert map card to Lit/TS

### PR DESCRIPTION
**Pull request by balloob**

### CODE REVIEW

File: src/panels/lovelace/cards/hui-map-card.ts [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+  public static getStubConfig() {
+    return { entities: [] };
+  }
+
+  @property() public hass?: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true })
+  public isPanel = false;
+
+  @property()
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-map-card.ts]: Can we decide on a standard for decorators? I don't like how we're doing inline sometimes but not always. I would vote for not inline as it is easier to be consistent and avoid long lines

File: src/panels/lovelace/cards/hui-map-card.ts [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+      }
+      this._leafletMap.invalidateSize();
+    },
+    100,
+    false
+  );
+  private _mapItems: Array<Marker | Circle> = [];
+  private _connected = false;
+
+  public setConfig(config): void {
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-map-card.ts]:
```suggestion
  public setConfig(config: MapCardConfig): void {
```

File: src/panels/lovelace/cards/hui-map-card.ts [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+      }
+
+      :host([ispanel]) #root {
+        height: 100%;
+      }
+    `;
+  }
+}
+
+customElements.define("hui-map-card", HuiMapCard);
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-map-card.ts]: use customElement decorator

File: src/panels/lovelace/editor/config-elements/hui-map-card-editor.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { EntitiesEditorEvent, EditorTarget } from "../types";
 import { HomeAssistant } from "../../../../types";
 import { LovelaceCardEditor } from "../../types";
 import { fireEvent } from "../../../../common/dom/fire_event";
-import { Config } from "../../cards/hui-alarm-panel-card";
+import { MapCardConfig } from "../../cards/hui-map-card";
```
Comment (iantrich) [inline src/panels/lovelace/editor/config-elements/hui-map-card-editor.ts]: oops 😄

**Review by iantrich — APPROVED** [review]
looks good after those changes are made and addressing the Travis error

`src/panels/lovelace/editor/config-elements/hui-map-card-editor.ts(66,5): error TS2322: Type '(string | EntityConfig)[]' is not assignable to type 'string[]'.`

File: src/panels/lovelace/cards/hui-map-card.ts (hunk 1 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-map-card.ts]: Well, prettier will put it on a new line if the property options get too long (as happened on the prop above this one).

**Pull request merged**